configuration_go/k8sutil/v2/builders.go

```diff
@@ -0,0 +1,132 @@
+package v2
+
+import (
+	"log"
+
+	"github.com/observatorium/observatorium/configuration_go/k8sutil"
+	corev1 "k8s.io/api/core/v1"
+)
+
+const (
+	defaultTerminationGracePeriodSeconds = 120
+	ServiceManifestKey                   = "service"
+	ServiceMonitorManifestKey            = "serviceMonitor"
+	ServiceAccountManifestKey            = "serviceAccount"
+	StatefulSetManifestKey               = "statefulSet"
+)
+
+type StatefulSetBuilder struct {
+	MetaConfig                    MetaConfig
+	Containers                    []ContainerProvider
```

Comment:
This sounds weird. A statefulset contains pods, not container. 🤔

Reply:
The idea with these "builders" is to standardise how StatefulSet or Deployments are created. It flattens and provides most important parameters for the Pod and StatefulSet and then takes care for building everything, from the Pod itself, the statefulSet and all resources expected by the containers. 
It does not do much, but I expect that the need will be the same for all components, and they won't need to implement this logic.
The alternatives are:
- this is implement in each component (in a simplified manner as here it is more generic)
- this is done by the caller that follows each step

Also the purpose of this builder is that it can be easily derived. You don't have to rebuild everything yourself chaining from the container, to final objects

Reply:
The builder has been removed.